executor/src/block_processor.rs

```diff
@@ -477,30 +477,35 @@ where
         let num_txns_to_skip = num_committed_txns - batch_first_version;
         if num_txns_to_skip != 0 {
             info!(
-                "Skipping the first {} transactions when committing",
-                num_txns_to_skip
+                "The lastest committed version: {}, the first version to commit: {}.\
+                 Skipping the first {} transactions and start committing from version {}",
+                num_committed_txns - 1, /* lastest_committed_version */
+                batch_first_version,
+                num_txns_to_skip,
+                batch_first_version + num_txns_to_skip
             );
         }
 
         // All transactions that need to go to storage. In the above example, this means all the
         // transactions in A, B and C whose status == TransactionStatus::Keep.
         let mut txns_to_commit = vec![];
         let mut num_accounts_created = 0;
-        for block in &block_batch.blocks {
-            for (txn, txn_data) in
-                itertools::zip_eq(&block.transactions, block.output.transaction_data())
-                    .skip(num_txns_to_skip as usize)
-            {
-                if let TransactionStatus::Keep(_) = txn_data.status() {
-                    txns_to_commit.push(TransactionToCommit::new(
-                        txn.clone(),
-                        txn_data.account_blobs().clone(),
-                        txn_data.events().to_vec(),
-                        txn_data.gas_used(),
-                        txn_data.status().vm_status().major_status,
-                    ));
-                    num_accounts_created += txn_data.num_account_created();
-                }
+        for (txn, txn_data) in block_batch
+            .blocks
+            .iter()
+            .map(|block| itertools::zip_eq(&block.transactions, block.output.transaction_data()))
+            .flatten()
+            .skip(num_txns_to_skip as usize)
```

Comment:
I'm not sure about this logic. You are skipping the first `num_txns_to_skip` transactions. Or do you want to skip the first `num_txns_to_skip` transactions whose status is `Keep`? Or something else? I just find it hard to understand.

Reply:
There are more than 1 problem. Let me force-update a newer change.